decidim-core/app/controllers/decidim/newsletters_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Decidim
+  # The controller to handle the user's notifications dashboard.
+  class NewslettersController < Decidim::ApplicationController
+    skip_authorization_check
+
+    layout "decidim/mailer"
+    helper Decidim::WidgetUrlsHelper
+    helper Decidim::SanitizeHelper
+
+    def show
+      @organization = current_organization
+      @newsletter = collection.find(params[:id])
```

Comment:
I'm unsure about this. Newsletters can have the `%{NAME}` special tag, which is replaced with the user name, so the newsletter looks customized. If we check for ID then we lose this customization when viewing it on the website.

What about hashing both the newsletter ID and the user name in a way that we can unhash it in the backend? This way we could send this hash as the ID, unhash it, get the newsletter ID and use the user name in the show page. 

Maybe we should discuss this with product on the related issue.